### PR TITLE
ARROW-5500: [R] read_csv_arrow() signature should match readr::read_csv()

### PR DESCRIPTION
**Pull request by nealrichardson**
This patch enumerates the various CSV parsing options and exposes them in an R-familiar way in the signature of `read_csv_arrow()`. It also adds a generic `read_delim_arrow()` for providing other delimiting characters, as well as a `read_tsv_arrow()`. In the process, I identified some limitations of the current reader (https://issues.apache.org/jira/browse/ARROW-5747) and of the R bindings to it (not yet ticketed), and added more docs and tests.

Other release-prep cleanup in here includes organization of the DESCRIPTION file, adding new functions to the pkgdown config, and adding a NEWS.md.

### CODE REVIEW

**Review by wesm — COMMENTED** [review]
Haven't looked carefully at the R code (could @romainfrancois take a look)? But I found the version number in NEWS.md to be misleading, since a CRAN release probably won't happen until after 0.14.0 goes out

File: r/NEWS.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# arrow 0.13.0.9000
```
[the remark this answers is not in the document]
Comment (nealrichardson) [reply r/NEWS.md]: It should. https://issues.apache.org/jira/browse/ARROW-5415 will update it to the release version in `dev/release/00-prepare.sh`. I have a branch for that based on this one.

[the remark this answers is not in the document]
Comment (nealrichardson) [reply r/NEWS.md]: https://github.com/nealrichardson/arrow/compare/readr-csv...nealrichardson:r-version-update

**Review by wesm — APPROVED** [review]
+1, modulo CI. Taking a look

**Comment by wesm** [comment]
Here's a passing Travis build: https://travis-ci.org/nealrichardson/arrow/builds/551482949

Appveyor running here: https://ci.appveyor.com/project/nealrichardson/arrow/builds/25596580. Will wait a bit yet

**Comment by wesm** [comment]
The R Appveyor build is running now so I'll merge once that passes

**Comment by wesm** [comment]
Build is passed

**Pull request closed without merging**